applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.cpp

```diff
@@ -405,41 +407,33 @@ void KratosGeoSettlement::PrepareModelPart(const Parameters& rSolverSettings)
         domain_part_names.emplace_back(sub_model_part.GetString());
     }
 
-    // Add nodes to computing model part
     std::set<Node::IndexType> node_id_set;
     for (const auto& name : domain_part_names) {
         auto& domain_part = main_model_part.GetSubModelPart(name);
-        for (const auto& node : domain_part.Nodes()) {
-            node_id_set.insert(node.Id());
-        }
+        GenericUtilities::CollectIdsFromEntity(domain_part.Nodes(), node_id_set);
```

Comment:
When changing the function to accept an output iterator, this would need to become:

```suggestion
        GenericUtilities::CollectIdsFromEntity(domain_part.Elements(),
                                               std::inserter(node_id_set, node_id_set.end()));
```